apis/core/v1alpha1/resource.go

```diff
@@ -44,6 +44,10 @@ const (
 	ResourceCredentialsSecretKubeconfigKey = "kubeconfig"
 )
 
+// LabelKeyProviderName is added to ProviderConfigUsages to relate them to their
+// ProviderConfig.
+const LabelKeyProviderName = "crossplane.io/provider-config"
```

Comment:
It feels weird to label the resource even if that information exists under spec but I understand that we do this to delegate the filterin to api-server instead of doing it via a predicate which makes the client do the filtering.

Reply:
What makes it weird?

This is done mostly for UX rather than anything else (though it does help with listing). i.e. `kubectl get providerconfigusage -l crossplane.io/provider-config=foo`

Reply:
It's a duplication of the information that already exists in the same object.

Reply:
I feel like that's not unidiomatic for Kubernetes - e.g node are automatically labelled by the Kubelet with labels that contain the same information as the node object to make it easier to select and match them.